contracts/MyToken.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/access/Ownable.sol";
+import "openzeppelin-solidity/contracts/security/Pausable.sol";
+
+contract MyToken is ERC20,Pausable,Ownable {
+
+    mapping (address => uint256) internal _balances;
+
+    constructor() ERC20("MyToken", "MK") {
+        uint256 balance =  1_000_000 * 10 ** decimals();
+        _mint(msg.sender,balance);  
+        _balances[msg.sender] = balance; 
+    } 
+
+    function decimals() public view virtual override returns (uint8) {
+        return 6;
+    }
+
+    function pause() public onlyOwner {
+        _pause();
+    }
+
+    function unpause() public onlyOwner {
+        _unpause();
+    }
+
+    function burn(address _from, uint256 _amount) public onlyOwner whenNotPaused{
```

Comment:
The burn function should not be called by the Owner, it is intended for burning tokens by the user.

Reply:
WhenNotPaused is not required, as the check in _beforeToken... will work